ruby_on_rails/advanced_forms_and_activerecord/installing_psql.md

```diff
@@ -0,0 +1,342 @@
+
+~~~bash
+psql
+~~~
+
+You should see the PostgreSQL prompt come up with the new role we just created, like so:
+
+~~~
+<role_name>=#
+~~~
+
+If you don't see a similar prompt, then reach out on [Discord](https://discord.gg/V75WSQG) for some help. 
+
+If you **do** see a similar prompt, then we can create a password for the role like so:
+
+~~~
+\password <role_name>
+~~~
+
+You'll be prompted to enter a password and to verify it. Once you are done, the prompt will return to normal. Now, we will configure the permissions for our new role:
+
+~~~
+grant all privileges on database <role_database_name> to <role_name>;
+~~~
+
+Remember that you should change the `<role_database_name>` and `<role_name>` (they should both the same)! If you see `GRANT` in response to the command, then you can type `\q` to exit the prompt.
+
+This role and password combination enables us to log in to the PostgreSQL service to perform operations, so let's save this into the environment so local Rails apps can do those things.
+
+#### 3.5 Saving Access Information in the Environment
+After finishing our configuration, the last step is save it into the environment to access later.
+
+In order to save our password to the environment, we can run this command:
+
+~~~bash
+echo 'export DATABASE_PASSWORD="<role_password>"' >> ~/.bashrc
+~~~
+
+Note here the name we've chosen for our environment variable: `DATABASE_PASSWORD`. Also, remember to update `<role_password>` in the command to what was set above!
+
+Now, this variable lives in our environment for us to use. As the variable is new, we'll want to reload the environment so that we can access it. To reload the environment, you can close and re-open your terminal.
+
+Once that's done, we can move to testing it out!
+
+</details>
+
+<details markdown="block">
+<summary class="dropDown-header">macOS
+</summary>
+
+### Step 1: Make sure the system is up to date
+Before running commands with homebrew, you'll want to make sure things are up to date. Run the following commands.
+
+~~~~bash
+brew update
+brew upgrade
+~~~~
+
+If your terminal doesn't recognize `brew`, then you'll need to go and install homebrew. You can find it and other installs in the curriculum [here](https://www.theodinproject.com/lessons/installation_lessons).
+
+### Step 2: Install the PostgreSQL packages
+Now that we've ensured our packages are up to date, we will use brew to install PostgreSQL.
+
+~~~bash
+brew install postgresql@14
+~~~
+
+After installation is complete, let's start the server using this command:
+
+~~~bash
+brew services start postgresql@14
+~~~
+
+If you are unsure about whether postgres is active, it's possible to check with this command:
+
+~~~bash
+brew services info postgresql@14
+~~~
+
+Got an error, or don't see an active service? Come visit the Discord for some help!
```

Comment:
Turn this into a link. There might be two or more occurrences of this that need attention but I won't be creating separate comments for them.

```suggestion
Got an error, or don't see an active service? Come visit the [Discord](https://discord.gg/V75WSQG) for some help!
```